extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/compress/CompressionTest.java

```diff
@@ -0,0 +1,67 @@
+package io.quarkus.resteasy.reactive.server.test.compress;
+
+import static io.restassured.RestAssured.get;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vertx.http.Compressed;
+import io.quarkus.vertx.http.Uncompressed;
+import io.restassured.response.ExtractableResponse;
+import io.restassured.response.Response;
+
+public class CompressionTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot(root -> root
+                    .addClasses(MyEndpoint.class)
+                    .addAsManifestResource(new StringAsset("bar"), "resources/file.txt")
+                    .addAsManifestResource(new StringAsset("doc"), "resources/my.doc"))
+            .overrideConfigKey("quarkus.http.enable-compression", "true");
+
+    @Test
+    public void testRoutes() {
+        assertCompressed("/endpoint/compressed");
+        assertUncompressed("/endpoint/uncompressed");
```

Comment:
Shouldn't we also assert the response body?

Reply:
Yes, we could.

Reply:
Done.